src/main/java/org/folio/circulation/services/ItemsInTransitReportService.java

```diff
@@ -79,9 +86,22 @@
   }
 
   private CompletableFuture<Result<ItemsInTransitReportContext>> fetchInstances(
-    Result<ItemsInTransitReportContext> context) {
+    ItemsInTransitReportContext context) {
 
-    return completedFuture(context);
+    var mapper = new InstanceMapper();
+    FindWithMultipleCqlIndexValues<Instance> fetcher
+      = findWithMultipleCqlIndexValues(instancesStorageClient, "instances", mapper::toDomain);
+
+    List<String> instanceIds = context.getItems().values()
+      .stream()
+      .map(Item::getInstanceId)
+      .filter(Objects::nonNull)
+      .distinct()
+      .collect(Collectors.toList());
```

Comment:
We'll need to do this quite often in this method - get some map from context, get some ids, remove nulls and collect distinct values. Can this be extracted to a method?

Reply:
I updated the code